lib/internal/modules/esm/loader.js

```diff
@@ -283,16 +384,16 @@ class DefaultModuleLoader {
    */
   forceLoadHooks() {}
```

Comment:
I think this needs to be updated to be part of the customization

Reply:
You're right. The other tests you added back to `main` caught this. Thanks!

Reply:
I don't understand exactly what this method does (and what, if anything, its corresponding implementation in `Hooks` should be), but the tests all pass now by having it call the `#customizations.forceLoadHooks()` method if set.

Reply:
I was trying to debug this and I noticed that the `myGlobal` test, the one failing as:

```shell
./node --no-warnings --experimental-loader 'data:text/javascript,export function globalPreload(){return "this.myGlobal=4"}' --print 'myGlobal'
```

Would pass if you did `echo 'console.log(myGlobal)' > entry.mjs` and ran:

```shell
./node --no-warnings --experimental-loader 'data:text/javascript,export function globalPreload(){return "this.myGlobal=4"}' ./entry.mjs
```

So maybe the `forceLoadHooks()` has something to do with getting the customizations registered for alternative code paths like those used for `--print`?